src/node/utils/caching_middleware.js

```diff
@@ -192,11 +191,12 @@ CachingMiddleware.prototype = new function () {
         res.write = old_res.write || res.write;
         res.end = old_res.end || res.end;
 
-        var headers = responseCache[cacheKey].headers;
+        let headers = {};
+        Object.assign(headers, (responseCache[cacheKey].headers || {}));
```

Comment:
Questions/comments:
* Is `responseCache[cacheKey]` guaranteed to be non-null here?
* I think the object spread operator is easier to read vs. `Object.assign()`.
* It would be good to add a comment explaining why a shallow copy is made.

Suggested change that addresses the above:
```suggestion
        const {[cacheKey]: {headers: cachedHeaders = {}} = {}} = responseCache;
        // Make a shallow copy of the cached headers so that we can add/change headers without
        // affecting the cache.
        const headers = {...cachedHeaders};
```